okhttp/src/main/java/okhttp3/OkHttpClient.java

```diff
@@ -159,7 +159,16 @@ private OkHttpClient(Builder builder) {
     this.cache = builder.cache;
     this.internalCache = builder.internalCache;
     this.socketFactory = builder.socketFactory;
-    if (builder.sslSocketFactory != null) {
+
+    boolean isTLS = true;
+    for (ConnectionSpec spec : connectionSpecs) {
+      if (connectionSpecs.equals(DEFAULT_CONNECTION_SPECS)) {
+        break;
+      }
```

Comment:
This check shouldn't be inside the for loop, unless you meant it to be something else.